OPUNetGameSelectWnd.cpp

```diff
@@ -322,11 +318,8 @@ void OPUNetGameSelectWnd::ClearGamesList()
 		if (SendDlgItemMessage(this->hWnd, IDC_GamesList, LVM_GETITEM, 0, (LPARAM)&item))
 		{
 			// Retrieve the HostedGameInfo pointer
-			 HostedGameInfo* hostedGameInfo = (HostedGameInfo*)item.lParam;
-			// Make sure the parameter was set
-			if (hostedGameInfo != nullptr) {
-				delete hostedGameInfo;	// Free the memory
-			}
+			HostedGameInfo* hostedGameInfo = (HostedGameInfo*)item.lParam;
+			delete hostedGameInfo;

```

Comment:
I don't understand what is going on here. It looks like creating a pointer to hostedGameInfo and then immediately deleting it. Pull request is good to merge as behaviour existed earlier, just seems weird to me.

Reply:
It's retrieving a pointer stored as a data field in the list view control. As the data element is an untyped 32-bit field, you must cast it back to the type of whatever you stored in there.
```cpp
// Get the item data
item.iItem = i;
if (SendDlgItemMessage(this->hWnd, IDC_GamesList, LVM_GETITEM, 0, (LPARAM)&item))
{
	// Retrieve the HostedGameInfo pointer
	HostedGameInfo* hostedGameInfo = (HostedGameInfo*)item.lParam;
	delete hostedGameInfo;
}
```